src/pages/DataSharingLanguageTool.js

```diff
@@ -0,0 +1,298 @@
+import {Styles} from "../libs/theme";
+import {br, button, div, h, input, label, span, textarea} from "react-hyperscript-helpers";
+import {RadioButton} from "../components/RadioButton";
+import AsyncSelect from "react-select/async/dist/react-select.esm";
+import {isNil} from "lodash/fp";
+import {Notifications, searchOntologies} from "../libs/utils";
+import {DataUseTranslation} from "../libs/dataUseTranslation";
+import {useState} from "react";
+
+const buttonStyle = { marginBottom: '2rem', color: '#777' };
+const labelStyle = { fontFamily: 'Montserrat', fontSize: '15px' };
+
+export default function DataSharingLanguageTool() {
+  const [general, setGeneral] = useState(false);
+  const [hmb, setHmb] = useState(false);
+  const [diseases, setDiseases] = useState(false);
+  const [ontologies, setOntologies] = useState();
+  const [other, setOther] = useState(false);
+  const [otherText, setOtherText] = useState();
+  const [nmds, setNmds] = useState(false);
+  const [gso, setGso] = useState(false);
+  const [pub, setPub] = useState(false);
+  const [col, setCol] = useState(false);
+  const [irb, setIrb] = useState(false);
+  const [gs, setGs] = useState(false);
+  const [mor, setMor] = useState(false);
+  const [npu, setNpu] = useState(false);
+  const [sdsl, setSdsl] = useState("");
+
+  const isTypeOfResearchValid = () => {
+    return (general || hmb || (diseases && !isNil(ontologies)) || other );
+  };
+
+  const clearOtherTextBox = () => {
+    document.getElementById('other_text').value = "";
+  };
+
+  const generate = () => {
+    isTypeOfResearchValid() ?
+      generateHelper()
+      : Notifications.showError({text: "Please complete question 1"});
+  };
+
+  const generateHelper = () => {
+    const darInfo = {
+      general: general, diseases: diseases, ontologies: ontologies, other: other, otherText: otherText,
+      hmb: hmb, methods: !nmds, forProfit: !npu
+    };
+    const summaries = [];
+    const dataUse = (DataUseTranslation.translateDarInfo(darInfo));
```

Comment:
@rushtong This method is not a good fit for the functionality of this page. It returns inaccurate and often conflicting language. it does not include any checks/language for most of the information checkboxes, as you can see in the darInfo constant it only includes methods which is the opposite of nmds and forProfit which is the opposite of npu. This causes issues, for example if someone doesn't check npu they get a statement that might not be true: "The dataset will be used in a study related to a commercial purpose" similarly if they don't check NMDS then they get this statement: "The primary purpose of the research is to develop and/or validate new methods for analyzing or interpreting data." which conflicts with the prior sentence that establishes the primary purpose to be either general, hmb, or disease related. Instead we should only use this method to generate the one sentence on primary purpose, and a different method to generate the sentences for all of the checkbox information. There already exists a method diseaseRestrictions in dataUsetranslation that does check for all of the information from the checkboxes, so I think we should use that to generate the secondary use restrictions

Reply:
I outlined the approach I am proposing in the code, what do you think?

Reply:
> DataUseTranslation.translateDarInfo(darInfo) is not a good fit for the functionality of this page. It returns inaccurate and often conflicting language.

If the above is true, then `translateDarInfo` has some serious problems. I'll look at this tomorrow. We should be using the exact same method for generating the language as that is the idea behind this entire page. If this doesn't generate the same language with the same questions, then it doesn't meet the requirements. The current functionality was approved by Jonathan, so we might need to loop him into this conversation.

Reply:
What we should be using is `DataUseTranslation.translateDataUseRestrictions` since this should be passing in a `dataUse` object that describes the constructed `dataset`. The reason we're getting confusing results is that using `translateDarInfo` is meant for Data Access Requests, not datasets.

Reply:
Just changed it to translate data use restrictions and the generated language is much better, more comprehensive and clear! I had to make a minor change to the method to accommodate the ontologies given, but it will not affect the functionality of the method elsewhere